packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -411,6 +411,86 @@ export class TokenBalancesController extends StaticIntervalPollingController<Tok
     }
   }
 
+  /**
+   * Get an Ethers.js Web3Provider for the requested chain.
+   *
+   * @param chainId - The chain id to get the provider for.
+   * @returns The provider for the given chain id.
+   */
+  #getProvider(chainId: Hex): Web3Provider {
+    return new Web3Provider(this.#getNetworkClient(chainId).provider);
+  }
+
+  /**
+   * Internal util: run `balanceOf` for an arbitrary set of account/token pairs.
+   *
+   * @param params - The parameters for the balance fetch.
+   * @param params.chainId - The chain id to fetch balances on.
+   * @param params.pairs - The account/token pairs to fetch balances for.
+   * @returns The balances for the given token addresses.
+   */
+  async #batchBalanceOf({
+    chainId,
+    pairs,
+  }: {
+    chainId: Hex;
+    pairs: { accountAddress: Hex; tokenAddress: Hex }[];
+  }): Promise<MulticallResult[]> {
+    if (!pairs.length) {
+      return [];
+    }
+
+    const provider = this.#getProvider(chainId);
+
+    const calls = pairs.map(({ accountAddress, tokenAddress }) => ({
+      contract: new Contract(tokenAddress, abiERC20, provider),
+      functionSignature: 'balanceOf(address)',
+      arguments: [accountAddress],
+    }));
+
+    return multicallOrFallback(calls, chainId, provider);
+  }
+
+  /**
+   * Returns ERC-20 balances for a single account on a single chain.
+   *
+   * @param params - The parameters for the balance fetch.
+   * @param params.chainId - The chain id to fetch balances on.
+   * @param params.accountAddress - The account address to fetch balances for.
+   * @param params.tokenAddresses - The token addresses to fetch balances for.
+   * @returns A mapping from token address to balance (hex) | null.
+   */
+  async getErc20Balances({
+    chainId,
+    accountAddress,
+    tokenAddresses,
+  }: {
+    chainId: Hex;
+    accountAddress: Hex;
+    tokenAddresses: Hex[];
+  }): Promise<Record<Hex, Hex | null>> {
+    // Return early if no token addresses provided
+    if (tokenAddresses.length === 0) {
```

Comment:
Makes sense to return early here; wondering if there is a specific use case you had in mind for adding the condition?
I think this fct will be used on clients to fetch balances on the fly for 1 or more tokens in the import flow; i , for now, dnt expect it to be called with an empty array; but makes sense to me to keep the check!